code/game/objects/items/reagent_containers/food/snacks.dm

```diff
@@ -2724,6 +2724,24 @@
 		icon_state = "[initial(icon_state)]-o"
 		package = FALSE
 
+/obj/item/reagent_container/food/snacks/wrapped/lampbar  //LAMP!
+	name = "LAMP Bar"
+	desc = "A glow in the dark candybar. Made with real glowshrooms. <B>\"LAMP!\"</B>"
+	icon_state = "lamp"
+	var/brightness_on = 3	//its not a very strong light
+	wrapper = /obj/item/trash/lamp
+	
+	
+/obj/item/reagent_container/food/snacks/wrapped/lampbar/New()
+	..()
+	reagents.add_reagent("nutriment", 4)
+	reagents.add_reagent("coco", 4)
+	reagents.add_reagent("mothpheromones", 4)
```

Comment:
`list_reagents = list("nutriment" = 4, "coco" = 4, "mothpheromones" = 4)`, on the path, not the New() proc.
Also you can set up the snack's tastes now. Example: `tastes = list("examples" = 2)`